package/MDAnalysis/coordinates/TRZ.py

```diff
@@ -588,6 +600,7 @@ def _write_next_frame(self, obj):
         out['p1a'], out['p1b'] = 20, 20
         out['nframe'] = ts.frame + 1  # TRZ wants 1 based
         out['ntrj'] = data['step']
+        out['natoms'] = self.n_atoms
```

Comment:
Could you also add a test for this too please?

Reply:
It's implicitly checked during `test_write_trajectory`. That's where I spot the bug in the first place, do we need to do it more explicitly?

Reply:
was it raising an error previously? If not we should explicitly check that the right n_atoms value is present in the file now.

Reply:
It will copy and write to a TRZ file, then read the written file for comparison. So it will raise an error during reading if the written `natoms` is wrong,

Reply:
Ok yes that works, thanks for walking me through it :)